Add connection quality related attributes to HomematicIP Cloud

SukramJ opened the pull request:
## Description:
This PR adds extra attributes to Homematic IP devices to determine connection quality, and helps to identify issues with it.

To keep the device as clean as possible these extra attributes are only available if parameter `show_extra_attributes` is set to `true `in config.

## Example entry for `configuration.yaml` (if applicable):
```yaml
homematicip_cloud:
  - accesspoint: IDENTIFIER
    authtoken: AUTHTOKEN
  - name: Location2
    accesspoint: IDENTIFIER2
    authtoken: AUTHTOKEN2
    show_extra_attributes: true
```
## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`.
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io/pull/8651)

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

SukramJ committed:
Added connection quality related attributes

SukramJ commented:
Hi @mxworm, please check my small PR.

mxworm commented:
Hi, Please only show useful information to the user. Keep the devices as clean as possible. If we could add a „debug“ switch to the Config->Integrations to enable/disable this additional info. Ciao

SukramJ committed:
removed attribute

attribute is already used in method <available>

SukramJ commented:
Hi, 
i am aware of keeping the infos to the user as clean as possible, as you already mentioned it in a ticket before. 

- I already saw that the `unavailable `attribute is not necessary, because it is already used in the <available> method. I've removed that.
- `dutyCycle` will be only displayed if a device reaches it dutyCycle and will not be usable for some time, so normally nothing will be shown. 
- `device_rssi`, `peer_rssi` only one (currently not sure which) will normally displayed for battary driven device. Only devices with permantent power supply show both values.

From my personal perspective these information are useful for a user when working with wireless devices. 
I already had trouble with rssi. EQ-3 told me that my problems with alarm are related to connection issues with a window sensor (bad rssi), but i had no chance to identify which sensor was effected. Also the vendor app hides these information.
Compared to homematic classic there are only a few attributes shown to a device. ;-)

Btw: I am not checking the whole component for missing attributes, but i check what is missing for me and could be helpful to others. 

I apprechiate your component and i am where thankful for your feedback, support and this component in homeassistant. This component is the main reason why i switched to home assistant!

Do still think its worth adding a debug switch to the configuraion for just one or somtimes two additional attribute normally shown?
I could do that.

BR 
Markus

SukramJ commented:
rssiDevice = HAP -> Device (is available for all wireless devices)
rssiPeer = Device -> HAP (is available for some wireless devices)

SukramJ commented:
Hi @mxworm,
after thinking a while about your proposal, i go for your idea to have an additional parameter to add more attributes. But instead of naming it  `debug` i thougt naming it `show_extra_attributes` is more applicable. 
Currently it is only added to configuration.yaml  is optional.
In a follow up PR this could be added to Config->Integrations (I have no clue how, but i am excited).

What do you think?

SukramJ committed:
added parameter for additional attributes

SukramJ commented:
Review comment from @MartinHjelmare for PR [Add Groups to Homematic IP](https://github.com/home-assistant/home-assistant/pull/21076):
> We don't want these kind of options in the config flow. The config flow is mainly for connecting and authentication. When we have an options flow for config entries, it can be implemented with that. For now, please remove the options from the config flow and just keep them in config yaml.

I think this means we should wait until an option flow exists and add the option to that flow.
Until then the parameter remains only in config yaml

SukramJ commented:
Hi @mxworm is here anything left to do for me?

mxworm commented:
Do you have a BROLL/FROLL device? (Did I ask you already?)

SukramJ commented:
I do not have either.
You haven't asked. ;-)

SukramJ committed:
reverted change
